app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CreateExpense.kt

```diff
@@ -0,0 +1,359 @@
+package com.github.se.wanderpals.ui.screens.trip.finance
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExposedDropdownMenuBox
+import androidx.compose.material3.ExtendedFloatingActionButton
+import androidx.compose.material3.FloatingActionButtonDefaults
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.toMutableStateList
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import com.github.se.wanderpals.model.data.Category
+import com.github.se.wanderpals.model.data.Expense
+import com.github.se.wanderpals.model.data.User
+import com.github.se.wanderpals.model.viewmodel.ExpenseViewModel
+import com.github.se.wanderpals.ui.navigation.NavigationActions
+import com.github.se.wanderpals.ui.screens.DateInteractionSource
+import com.github.se.wanderpals.ui.screens.MyDatePickerDialog
+import com.github.se.wanderpals.ui.theme.WanderPalsTheme
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+/**
```

Comment:
Since this composable is quit big, it could be nice to provide a more detailed description of what is the purpose of the create expense screen. For example, the detailed description could start with something like this : "The user can select the amount he has to pay and for which user he has paid...." This is just an example.

Also a little note, I think it is good here to not split composables in this file in other files since it is "only the expense screen", however what I personally like to do is to add a little  comment to just indicate what the composable represents. For example here I would have added //Expense amount, //Expense date at the top of each composable so that when someone reads the code, he will see with the comment what the composable represents. But this is just a personal preference when I read code.